src/molgenis_fdp_harvester/ckan_harvest/baseparser.py

```diff
@@ -0,0 +1,1158 @@
+# SPDX-FileCopyrightText: Open Knowlege
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# This material is copyright (c) Open Knowledge.
+# It is open and licensed under the GNU Affero General Public License (AGPL) v3.0
+# Original location of file: https://github.com/ckan/ckanext-dcat/blob/master/ckanext/dcat/profiles/base.py
+#
+# Modified by Stichting Health-RI to remove dependencies on CKAN
+
+import datetime
+import json
+from urllib.parse import quote
+import re
+
+from dateutil.parser import parse as parse_date
+from rdflib import term, URIRef, BNode, Literal
+from rdflib.namespace import Namespace, RDF, XSD, SKOS, RDFS
+from geomet import wkt, InvalidGeoJSONException
+
+# from ckantoolkit import config, url_for, asbool, get_action
+# from ckan.model.license import LicenseRegister
+# from ckan.lib.helpers import resource_formats
+# from ckanext.dcat.utils import DCAT_EXPOSE_SUBCATALOGS
+
+DCT = Namespace("http://purl.org/dc/terms/")
+DCAT = Namespace("http://www.w3.org/ns/dcat#")
+DCATAP = Namespace("http://data.europa.eu/r5r/")
+ADMS = Namespace("http://www.w3.org/ns/adms#")
+VCARD = Namespace("http://www.w3.org/2006/vcard/ns#")
+FOAF = Namespace("http://xmlns.com/foaf/0.1/")
+SCHEMA = Namespace("http://schema.org/")
+TIME = Namespace("http://www.w3.org/2006/time")
+LOCN = Namespace("http://www.w3.org/ns/locn#")
+GSP = Namespace("http://www.opengis.net/ont/geosparql#")
+OWL = Namespace("http://www.w3.org/2002/07/owl#")
+SPDX = Namespace("http://spdx.org/rdf/terms#")
+
+namespaces = {
+    "dct": DCT,
+    "dcat": DCAT,
+    "dcatap": DCATAP,
+    "adms": ADMS,
+    "vcard": VCARD,
+    "foaf": FOAF,
+    "schema": SCHEMA,
+    "time": TIME,
+    "skos": SKOS,
+    "locn": LOCN,
+    "gsp": GSP,
+    "owl": OWL,
+    "spdx": SPDX,
+}
+
+PREFIX_MAILTO = "mailto:"
+
+GEOJSON_IMT = "https://www.iana.org/assignments/media-types/application/vnd.geo+json"
+
+HARVEST_DEFAULT_LABEL_LANGUAGE = "en"
+
+
+# Below taken from https://github.com/ckan/ckan/blob/master/ckan/lib/munge.py
+def _munge_to_length(string: str, min_length: int, max_length: int) -> str:
+    """Pad/truncates a string"""
+    if len(string) < min_length:
+        string += "_" * (min_length - len(string))
+    if len(string) > max_length:
+        string = string[:max_length]
+    return string
+
+
+def substitute_ascii_equivalents(text_unicode: str) -> str:
```

Comment:
Please use libraries for this stuff. For example 

from unidecode import unidecode

text_unicode = "Mëtàl Hëàd"
text_ascii = unidecode(text_unicode)
print(text_ascii)  # Outputs: "Metal Head

using the pip install unidecode

Reply:
Agreed it could be better to use a library, but do we want to deviate from upstream on this?

Reply:
deze code is meer dan 20 jaar oud. weet niet of het daar snel veranderd ;-)  maar kan een pull request in de bron aanmaken

Reply:
Removed the code